docs/rfcs/output-field-name-semantic.md

```diff
@@ -0,0 +1,236 @@
+# Datafusion output field name semantic
```

Comment:
Might be helpful to number/add a date to the RFCs? E.g.

`2021-05-25-output-field-name-semantic.md` 
or
`00001-output-field-name-semantic.md `